design-spec/priviledge-decentralized-swupdates.tex

```diff
@@ -177,32 +177,483 @@ \section{Design}
+    have a way to determine when a sufficient majority of them have upgraded to
+    the new version.
+    % NOTE: might want to call these software and parameters protocol updates,
+    % however this seems to be too specific to the protocol. What happens if in
+    % the future we add the ability to upgrade the ledger rules dynamically?
+  \item[Non-signaled] An update is said to be \emph{non-signaled} if the nodes
+    can move to the new protocol version without needing to know if other nodes
+    are ready to move to said version as well. In this kind of updates we assume
+    the nodes are already running a version of the software that can switch to
+    the new protocol. An example of non-signaled protocol updates is
+    \emph{parameters-update}. In a parameters-update, the nodes have already the
+    logic for handling the parameters, and only the values of these parameters
+    change after an update is activated.
+  \end{description}
+\item[Non-protocol] These updates do not affect the consensus rules. The current
+  design does not restrict the nature of these updates. It is up to the ledger
+  layer to react to these updates accordingly. They might include:
+  \begin{itemize}
+  \item updates to the software that runs on the blockchain nodes, which does
+    not affect the protocol. Examples of such updates might include performance
+    or usability improvements.
+  \item transfer of funds. For instance an update proposal might involve funding
+    the development of a certain feature.
+  \item updates to components of the blockchain ecosystem.
+  \end{itemize}
+\end{description}
+
+Note that, according to the preceding definition, an update that requires the
+node to download and install new software to run a \emph{new protocol}, and that
+\emph{also} updates protocol parameters is considered a signaled protocol
+update.
+
+% TODO: mention somewhere that protocol updates must be activated one at a time,
+% whereas non-protocol updates can be approved in parallel.
 
 \subsection{The lifecycle of a decentralized update}
 \label{sec:phases-an-update}
 % FROM 3 The lifecycle of a decentralized software update
 
-% Why do we need to have a governance system that captures the whole life cycle
+The update protocol described in this document aims at capturing the whole life
+cycle of an update.
+% TODO: why do we need to have a governance system that captures the whole life cycle
 % of an update? So for instance, why not supporting update activations only?
-
-\subsection{Voting and delegation}
-\label{sec:voting-delegation}
-% FROM 4 Update governance
-
-\subsubsection{Delegation}
+%
+% Why distinguish phases at all? SIP and implementations can all be proposals.
+% They only differ in that some kinds of approved implementations can be
+% activated.
+
+We distinguish the following \emph{phases} in the life cycle of an update:
+\begin{description}
+\item[Ideation] This is the phase where an update proposal is submitted to the
```

Comment:
Should you  mention that a SIP is submitted?
where a SIP is a text document in a structured form (e.g., JSON) that formalizes the minimum required for the description of a proposal.

Reply:
See if this is addressed in the latest version.